inc/logger.h

```diff
@@ -0,0 +1,43 @@
+#ifndef LOGGER_H
+#define LOGGER_H
+#include <QObject>
+#include <QString>
+#include <QDateTime>
+#include <QtGlobal>
+#include <QDebug>
+#include <QFile>
+#include <QTextStream>
+
+
+class Logger : public QObject {
+    Q_OBJECT
+
+public:
+    enum outputType {
```

Comment:
Rather than using an enum, we can use polymorphism plus generators to achieve this functionality in a bit of a nicer way. 

You would keep most of the implmemtation the same way, but add 3 abstract protected virtual functions called writeInfo, writeWarning, and writeError which are called by the slots that you have implemented. You then write two derived classes which implement those functions (one for file logging and one for console logging). 

For the Generators, you can write two static functions for Logger: `CreateFileLogger and CreateConsoleLogger) which respectively call the constructors for the derived classes and then return a pointer cast as a generic Logger* pointer. 

This ask might be a bit confusing so please reach out if anything doesn't make sense.